contents/blog/the-posthog-array-1-41-0.md

```diff
@@ -0,0 +1,158 @@
+- New: Change your email yourself
+- New: Hedgehog mode
+- Improved: Recordings interface
+- One more thing: Site apps 
+
+### New: Persons on events on by default
+
+We used to store events in one table and persons in another table. That meant that, once you reached billion event scale, any query which touched person properties would time out. But, no longer! After running [an extensive beta since update 1.39.0](/blog/the-posthog-array-1-39-0#beta-improving-query-performance-by-combining-persons-and-events), we've now added person data onto the events themselves. This change also applies to groups, adding group properties and aggregation to the same table. 
+
+You won’t see any UI changes as a result of this change — persons will still have their own Persons & Groups section on the sidebar, for example — but you will notice results are a lot (up to 400%!) faster for any queries involving persons, groups or events. This is a massive change, so be sure to [read the full announcement for more info](/blog/persons-on-events).
+
+> **Note:** As a result of putting persons on events we now have to be strict about identified and unidentified users. Now, only anonymous users can be aliased into other users - meaning that once you call `identify` on a user, you can no longer alias the user into another identified user. Because of that, we recommend double checking your `identify` and `alias` calls, and following advice above [how to upgrade a self-hosted instance to 1.41.0](/blog/the-posthog-array-1-41-0). 
+
+### New: Count of events per user
+
+![count events per user](../images/blog/array/1-41-0-count-events-per-user.png)
+
+Have you ever asked yourself '*What's the average average number of purchases per user?*' or '*What's the maximum number of forms submitted per user?'* 
+
+Questions like these used to be hard to answer with PostHog, but no more! We've added a new series option for insights, enabling you to see the count of events per user - including sub-options for seeing the averages, minimum and maximum count of events per user. 
+
+**Bonus update:** While working on this feature, we also took the opportunity to improve our math selector UI.
+
+### New: Text cards on dashboards
+
+![text cards on dashboards](../images/blog/array/1-41-0-text-card-on-dashboard.png)
+
+Previously there was no easy way to add context or links to a dashboard, meaning you may have to send long explanations when sharing a dashboard with teammates. That's why we've added the option for users on paid plans to add text cards where they can add any information they want, including metadata, images or gifs!
+
+### New: App metrics
+
+![app metrics](../images/blog/array/1-41-0-app-metrics-insight.png)
+
+Curious how well your apps are doing? Previously, you may have had to pour over the AWS logs, but now you can head to the new app metrics page to find out how many events an app has processed, how many retries were attempted and what errors may have occured. Very handy. 
```

Comment:
```suggestion
Curious how well your apps are doing? Previously, you may have had to pour over the AWS logs, but now you can head to the new app metrics page to find out how many events an app has processed and what errors may have occurred. Very handy. 
```

Reply:
cc @joethreepwood does the suggestion above make sense wording-wise? This feature is _not_ generally available.

Reply:
```suggestion
Curious how well your apps are doing? Previously, you may have had to pour over the AWS logs, but now you can head to the new app metrics page to find out how many events an app has processed, how many retries were attempted and what errors may have occured. Very handy. 

App metrics are only available for users on Scale or Enterprise plans. 
```